velox/common/base/SimdUtil-inl.h

```diff
@@ -1436,4 +1436,146 @@ inline bool memEqualUnsafe(const void* x, const void* y, int32_t size) {
   return true;
 }
 
+namespace detail {
+
+/// NOTE: SSE_4_2`s the performance of simdStrStr is a little slower than
```

Comment:
Can you use 32 bytes register if AVX2 is available?  We can also extend the dispatch on `k` to 34 as well in that case.

Reply:
>Can you use 32 bytes register if AVX2 is available?

I think xsimd api use 32 bytes when avx2 is available :

```
        template <class A>
        inline batch<double, A> load_unaligned(double const* mem, convert<double>, requires_arch<avx>) noexcept
        {
            return _mm256_loadu_pd(mem);
        }
```